src/main/java/org/openrewrite/hibernate/ReplaceLazyCollectionAnnotation.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2024 Schweizerische Bundesbahnen SBB
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.hibernate;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JLeftPadded;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.TypeUtils;
+import org.openrewrite.marker.Markers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class ReplaceLazyCollectionAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Replace @LazyCollection with jakarta.persistence.FetchType";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds the FetchType to jakarta annotations and deletes @LazyCollection.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+                maybeAddImport("jakarta.persistence.FetchType");
+                maybeRemoveImport("org.hibernate.annotations.LazyCollection");
+                maybeRemoveImport("org.hibernate.annotations.LazyCollectionOption");
+                return super.visitCompilationUnit(cu, executionContext);
+            }
+
+            @Override
```

Comment:
This looks wrong, doesn't it?

Reply:
This looks off indeed 🤔 We use [code-suggester](https://github.com/googleapis/code-suggester) to turn a diff into these comments; seems like it's off here somehow.

Reply:
What is should have done is (only) suggest that you use `ctx` instead of `executionContext` for that variable, as per our code convention. We'd hoped adding this automation would make that as easy as a one click apply here, but unfortunately not (yet).

Reply:
Renamed all occurrences to `ctx`